server/core/region_storage.go

```diff
@@ -96,6 +104,10 @@ func (s *RegionStorage) backgroundFlush() {
 
 // SaveRegion saves one region to storage.
 func (s *RegionStorage) SaveRegion(region *metapb.Region) error {
+	err := encryption.EncryptRegion(region, s.encryptionKeyManager)
```

Comment:
It seems that this will change the value of the `region *metapb.Region`. I think this is not the expected behavior.